cirq/google/serializable_gate_set.py

```diff
@@ -51,12 +49,35 @@ def __init__(self, gate_set_name: str,
                 forms of gates to GateOperations.
         """
         self.gate_set_name = gate_set_name
-        self.serializers = defaultdict(
-            list)  # type: Dict[Type, List[op_serializer.GateOpSerializer]]
+        self.serializers: Dict[Type, List[op_serializer.GateOpSerializer]] = {}
         for s in serializers:
-            self.serializers[s.gate_type].append(s)
+            self.serializers.setdefault(s.gate_type, []).append(s)
         self.deserializers = {d.serialized_gate_id: d for d in deserializers}
 
+    def with_gates(
```

Comment:
Rename this to `with_more_gates` or something similar, to make it clearer that the original gates are not being discarded.

Reply:
Changed to `with_added_gates`.